proseco/core.py

```diff
@@ -568,31 +568,33 @@ def dark(self):
         if self.dark_date:
             dark_date = self.dark_date
             select = 'nearest'
+            dark_id = get_dark_cal_id(date=dark_date, select=select)
+
+            # Warn if the pickled dark_date is not the same as we would have gotten based
+            # on the observation date.
+            dark_id_for_date = get_dark_cal_id(date=self.date, select='before')
```

Comment:
With regard to testing, I was trying to just move around self.date, pickle, and re-load. The only confusion there I have is that we do have a lot of dates (aca.acqs.date, aca.guides.date, aca.fids.date, aca.date).  I don't recall if these only exist as an artifact of using this class for everything.

Reply:
Well that `dates` thing is a feature of using a class structure where acq, guide, and fid selection are viewed as separate processes so each one needs to know its date.  So in theory you can change them all independently.

Reply:
Yeah, I was trying to remember if we had a real use case for being able to independently set date or if it was just due to feature (aka artifact) of the class .  Fine either way, just harder to define a test that fiddles with it if we wanted to do so.

Reply:
I defined a test that fiddles with the date and it seems to work fine.  Each class just cares about its date.  The new test is (I think) a whittled down version of the worst-case scenario you gave about a dark cal mismatch from prelim.  With any luck if/when that happens there will be a visible UserWarning.